src/globus_sdk/services/search/data.py

```diff
@@ -215,9 +212,11 @@ def add_sort(
         :param additional_fields: additional data to include in the sort document
         """
         self["sort"] = self.get("sort", [])
-        sort = {"field_name": field_name, **(additional_fields or {})}
-        if order is not None:
-            sort["order"] = order
+        sort = {
+            "field_name": field_name,
+            "order": order,
+            **(additional_fields or {}),
+        }
```

Comment:
This is an interesting case -- `field_name` is a mandatory method parameter, but `order` is not. Therefore, `order` could be overwritten by `additional_fields`, but `field_name` probably should not be.

```suggestion
        sort = {
            "order": order,
            **(additional_fields or {}),
            "field_name": field_name,
        }
```

Reply:
I'd argue in favor of the current proposed version on this one (`**` expansion at the end).

`query.add_sort("foo", additional_fields={"field_name": "bar"})` is an inherently ambiguous request. I don't think we need to guarantee some well-defined behavior in such a case. I think such usage should be -- intentionally -- undefined in behavior, and at best consistent and deterministic on a given SDK version.

So I like putting `**` expansion at the end as a consistent style element for the SDK.

---

I could be convinced that the other is better, but I think I'd need to see what the usage we're trying to support is.

Reply:
I think we need to answer the broader question here about how we want these 'escape valve' params (`additional_fields` and `query_params`) to behave. Should they always take or give precedence to the explicitly passed params or just leave the behavior undefined and up to the specific method implementation.

I think that the current behavior in the SDK of leaving it undefined is not great. For example, this is what `add_sort` is doing currently:
```
sort = {"field_name": field_name, **(additional_fields or {})}
if order is not None:
    sort["order"] = order
```
which can lead to some odd behavior where something like this:
```
SearchQuery().add_sort("field1", order="asc", additional_fields={
         "field_name": "field2",
         "order": "desc",
    }
)
```
results in `field_name="field2"` and `order="asc"`.
I think this is incredibly unintuitive even if it was a bad query to begin with.

All this too say, I agree with Stephen and like the idea of "putting ** expansions at the end as a consistent style element for the SDK". Curious to hear your thoughts though...